src/daft-functions/src/tokenize/decode.rs

```diff
@@ -0,0 +1,126 @@
+use std::sync::Arc;
+
+use common_error::{DaftError, DaftResult};
+
+use daft_core::{
+    array::{ops::as_arrow::AsArrow, ListArray},
+    datatypes::{Field, Utf8Array},
+    schema::Schema,
+    DataType, IntoSeries, Series,
+};
+use daft_dsl::{functions::ScalarUDF, ExprRef};
+use daft_io::IOConfig;
+use serde::Serialize;
+
+use crate::tokenize::bpe::DaftBPE;
+
+// Helper function that operates on a single Utf8 series
+fn decode_list(series: &Series, bpe: &DaftBPE) -> DaftResult<String> {
+    if !series.data_type().is_integer() {
+        return Err(DaftError::TypeError(format!(
+            "expected integer list inner type, got {}",
+            series.data_type()
+        )));
+    }
+    let series = series.cast(&DataType::UInt32)?;
+    let tokens: Vec<u32> = series.u32()?.as_arrow().values_iter().copied().collect();
```

Comment:
we probably don't need to copy the values here as the decoder takes in a reference.

In a [separate PR](https://github.com/Eventual-Inc/Daft/pull/2526/files#diff-f1df94618328daee0642d1f7432bc8caf8099df4d54709ca56377a1b250e78b3R363), i added a helper method to get the values as a slice, but you should be able to do that manually 


```rs
let u32_arr = series.u32()?.as_arrow();
let data: &PrimitiveArray<u32> = u32_arr.data().as_any().downcast_ref().unwrap();

let tokens: &[u32] = data.values().as_slice():
```